src/services/histories/sync/postgresql.py

```diff
@@ -0,0 +1,139 @@
+            self.__is_connected = True
+            self._create_table_if_not_exists()
+        except Exception as e:
+            self.__is_connected = False
+            logger.error(f'Connection Error: {str(e)}')
+
+    @exception_handler
+    def sync(self):
+        logger.info('PostgreSQL sync has is been called')
+        self.__wires_plat = WiresPlatModel.find_one()
+        if not self.__wires_plat:
+            logger.error("Please add wires-plat")
+        else:
+            self._sync()
+
+    def _sync(self):
+        store = []
+        for point in PointModel.find_all():
+            point_last_sync_id: int = self._get_point_last_sync_id(point.uuid)
+            for psh in PointStoreHistoryModel.get_all_after(point_last_sync_id, point.uuid):
+                point_store_history: PointStoreHistoryModel = psh
+                point: PointModel = point_store_history.point
+                query_details = f'INSERT INTO {self.config.table_name}_details (point_uuid,client_id,site_id,' \
+                                f'device_id,point_name,device_name,network_name,driver) VALUES (%s, %s, %s,' \
+                                f'%s, %s ,%s, %s, %s) ON CONFLICT (point_uuid) DO NOTHING'
+                data_details = (point_store_history.point_uuid, self.__wires_plat.client_id, self.__wires_plat.site_id,
+                                self.__wires_plat.device_id, point.name, point.device.name, point.device.network.name,
+                                point.driver)
+                query = f'INSERT INTO {self.config.table_name} (id,point_uuid,value,value_original,value_raw,fault,' \
+                        f'fault_message,ts_value,ts_fault) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s)'
+                data = (point_store_history.id, point_store_history.point_uuid, point_store_history.value,
+                        point_store_history.value_original, point_store_history.value_raw, point_store_history.fault,
+                        point_store_history.fault_message, point_store_history.ts_value, point_store_history.ts_fault)
+                with self.__client:
+                    with self.__client.cursor() as curs:
+                        try:
+                            curs.execute(query_details, data_details)
+                            curs.execute(query, data)
+                        except psycopg2.Error as e:
+                            logger.error(str(e))
+                row = {
+                    'table': self.config.table_name,
+                    'time': point_store_history.ts_value,
+                    'details': data_details,
+                    'fields': point_store_history
+                }
+                store.append(row)
+        if len(store):
+            logger.debug(f"Storing: {store}")
+            self.__client.write_points(store)
+            logger.info(f'Stored {len(store)} rows on {self.config.table_name} table')
```

Comment:
These are done for bulk insertion on InfluxDB, it should show exceptions but exceptions are handled so app is not crashing.